ui/raidboss/popup-text.ts

```diff
@@ -380,17 +380,55 @@ class TriggerOutputProxy {
     }
 
     return value.replace(/\${\s*([^}\s]+)\s*}/g, (_fullMatch: string, key: string) => {
-      if (params !== undefined && key in params) {
-        const str = params[key];
-        switch (typeof str) {
-          case 'string':
-            return str;
-          case 'number':
-            return str.toString();
+      if (params !== undefined) {
+        let prop: string | undefined = undefined;
+
+        if (!(key in params) && key.includes('.')) {
+          const parts = key.split('.');
+          key = parts[0] ?? '';
+          prop = parts[1];
+        }
+
+        if (key in params) {
+          const val = params[key];
+          switch (typeof val) {
```

Comment:
I don't see any way to make this much easier.  I do think probably `if` statements are more easy to read than nested `switch` statements.  (And probably also more early outs rather than super nested code.)  But, I think this is fine if there's only two levels here.  If you thought we wanted three, then probably a recursive function is needed that point.